### DISCUSSION TITLE
[Beta] UI ordering of componets different

### DISCUSSION THREAD

**Discussion by julianstolp** [netbox-community/netbox#10955] [Q&A] [answered]
Hello, i tried the Beta today and was just wondering why the ordering at the device view is different? 
Is this by design or a bug? 
Journal and Changelog second and third place, imho the least significant tabs.

v3.4-beta1: https://beta-demo.netbox.dev/dcim/devices/19/
v3.3.8 https://demo.netbox.dev/dcim/devices/19/

![Untitled](https://user-images.githubusercontent.com/41152397/202480017-7f26008c-7f4f-4468-a602-3adb01dfabe2.png)

**Comment by MajesticFalcon** [comment]
Probably has something to do with https://github.com/netbox-community/netbox/issues/9072

**Comment by kkthxbye-code** [comment]
It's related to https://github.com/netbox-community/netbox/issues/9072 / https://github.com/netbox-community/netbox/commit/0d7851ed9de2792ea6d9ed223c315c235290ddd7

And I agree that it's a little jarring. @jeremystretch did you have any thoughts about ordering when implementing this? I think it makes sense for device componenents to come before journal/changelog/config context.

**Comment by jeremystretch** [comment, marked as answer]
I actually prefer it this way, as the arrangement of tabs is consistent for all objects.

**Reply by candlerb** [reply]
On a device, "Interfaces" are the thing I reach for 95% of the time.  Followed by other components.  Things like Config Context, very rarely.

Therefore, having them ordered left-to-right from "important" to "less important" is better for me, just because that's the order I scan them visually, although I accept that is a subjective judgement.

If there is a consistency issue, you could make "Journal" and "Changelog" consistently the right-most tab on all items.

**Reply by mtinberg** [reply]
I think the consistency that is achieved here is `[object,fixed-tabs],[optional-tabs,...]` so all the fixed tabs which are always present are always in the same position and the variable tabs then sort off to the right.  How the variable tabs are sorted maybe a question, whether it's just alphabetic or has some weight/priority value.  That probably means there can be a consistent way to template n-number of variable tabs without duplicating logic for built-in variable tabs vs. add-on variable tabs, which may make the templates easier to understand.

But that's just a guess, I didn't go spelunking to try and figure it out for sure.
